docs/rules-engine/how-to-use/industrialize-ruleset-generation.md

```diff
@@ -0,0 +1,295 @@
+
+Note that if you want the operand to match any simple type `string | boolean | Date | number`, you can use the 
+`SupportedSimpleTypes` interface. It will be replaced with this list of types during the metadata extraction.
+
+Today, we do not yet support references to aliases from different types, so the `SupportedSimpleTypes` will be the only
+alias you can use to refer to the `string | boolean | Date | number` type:
+
+```typescript
+export const notArrayContains: Operator<SupportedSimpleTypes[], SupportedSimpleTypes> = {
+  name: 'notArrayContains',
+  evaluator: (array, value) => array.indexOf(value) === -1,
+  validateLhs: (inputArray) => Array.isArray(inputArray)
+};
+```
+
+### Facts Metadata
+
+The fact files are identified thanks to the `factFilePatterns` parameters of the `@o3r/rules-engine:extractor` builder.
+The extractor will extract all the facts models (identified with the `FactDefinitions` interface) of your project:
+
+```typescript
+const EnumType = 'a' | 'b';
+
+interface OtherInterface {
+  /**
+   * Description of property
+   */
+  property: number
+}
+interface ComplexInterface  {
+  /**
+   * Description of property1
+   */
+  property1: OtherInterface[]
+}
+
+interface CustomFacts extends FactDefinitions {
+  /** Description of fact1 */
+  fact1: string,
+  /** Description of fact2 */
+  fact2: ComplexInterface,
+  /** Description of fact3 */
+  fact3: EnumType
+}
+```
+```json5
+// rules.facts.metadata.json
+{
+  "facts": [
+    {
+      "name": "fact1",
+      "description": "Description of fact1",
+      "type": "string"
+    },
+    {
+      "name": "fact2",
+      "description": "Description of fact2",
+      "type": "object",
+      "schemaFile": "fact-schemas/complex-interface.schema.json"
+    },
+    {
+      "name": "fact3",
+      "description": "Description of fact3",
+      "type": "string",
+      "enum": ["a", "b"]
+    }
+  ]
+}
+```
+
+You can note that the `ComplexInterface` fact results in a separated schema file that validates the interface model:
+
+```json5 
+{
+  "type": "object",
+  "properties": {
+    "property1": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/OtherInterface"
+      }
+    }
+  },
+  "definitions": {
+    "OtherInterface": {
+      "type": "object",
+      "description": "Description of property1",
+      "properties": {
+        "property": {
+          "description": "Description of property",
+          "type": "number"
+        }
+      }
+    }
+  }
+}
+```
+
+> [!NOTE]
+> The models for the `OperatorsMetadata` are available in the [rules engine metadata interface file](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/rules-engine/builders/rules-engine-extractor/helpers/rules-engine.extractor.interfaces.ts).
+
+### Project with module sub-entries
+
+As mentioned in the previous section, the extractor looks into your project and the files of its dependencies to identify
+the schema of your complex facts.
```

Comment:
formatting

Reply:
This is markdown, the back to line is just to avoid too long lines on the file. This won't appear in the documentation.